server/src/main/java/org/apache/druid/segment/realtime/appenderator/SinkQuerySegmentWalker.java

```diff
@@ -415,7 +453,110 @@ public static String makeHydrantCacheIdentifier(final SegmentId segmentId, final
     // with subsegments (hydrants).
     return segmentId + "_H" + hydrantNumber;
   }
+  
+  /**
+   * Emit query/segment/time, query/wait/time and query/segmentAndCache/Time metrics for a Sink.
+   * It accumulates query/segment/time and query/segmentAndCache/time metric for each FireHydrant at the level of Sink.
+   * query/wait/time metric is the time taken to process the first FireHydrant for the Sink.
```

Comment:
Please extend the javadoc here in two ways:

- Link to `MetricsEmittingQueryRunner` and point out that this is derived from that class.
- Explain that the class behaves differently based on whether `segmentId` is null or nonnull. When nonnull it's in "accumulate" mode and when null it's in "emit" mode.

Reply:
Have made the change.